pkg/cloud/services/networking/network.go

```diff
@@ -147,6 +150,33 @@ func (s *Service) ReconcileNetwork(openStackCluster *infrav1.OpenStackCluster, c
 	return nil
 }
 
+func (s *Service) DeletePorts(openStackCluster *infrav1.OpenStackCluster) error {
+	networkID := openStackCluster.Spec.Network.ID
+	portList, err := s.client.ListPort(ports.ListOpts{
+		NetworkID:   networkID,
+		DeviceOwner: "",
+	})
+	if err != nil {
+		if capoerrors.IsNotFound(err) {
+			return nil
+		}
+		return fmt.Errorf("searching for no owner port error: %v", err)
```

Comment:
The error message is more clear if the actual request type is part of the message, e.g. "list".
```suggestion
		return fmt.Errorf("list ports of network %q: %v", networkID, err)
```